src/qt/paymentserver.cpp

```diff
@@ -241,14 +241,10 @@ void PaymentServer::ipcParseCommandLine(int argc, char* argv[])
             PaymentRequestPlus request;
             if (readPaymentRequestFromFile(arg, request))
             {
-                if (request.getDetails().network() == "main")
-                {
-                    SelectParams(CBaseChainParams::MAIN);
-                }
-                else if (request.getDetails().network() == "test")
-                {
-                    SelectParams(CBaseChainParams::TESTNET);
-                }
+                // TODO find out why this excludes regtest
```

Comment:
Maybe that's the reason, but why not support it for free then?
I can further simplify the code but not making a special case of regtest here. Besides, if it needs to be a special case then we need another field in chainparams, not treating regtest specially without doing so explicitly.